src/anemoi/training/train/forecaster.py

```diff
@@ -263,16 +319,59 @@ def calculate_val_metrics(
         y: torch.Tensor,
         rollout_step: int,
         enable_plot: bool = False,
-    ) -> tuple[dict, list]:
+    ) -> tuple[dict, list[torch.Tensor]]:
+        """Calculate metrics on the validation output.
+
+        Args:
+            y_pred: torch.Tensor
+                Predicted ensemble
+            y: torch.Tensor
+                Ground truth (target).
+            rollout_step: int
+                Rollout step
+            enable_plot: bool, defaults to False
+                Generate plots
+
+        Returns
+        -------
+            val_metrics, preds:
+                validation metrics and predictions
+        """
         metrics = {}
         y_preds = []
         y_postprocessed = self.model.post_processors(y, in_place=False)
         y_pred_postprocessed = self.model.post_processors(y_pred, in_place=False)
-        for mkey, indices in self.metric_ranges_validation.items():
-            metrics[f"{mkey}_{rollout_step + 1}"] = self.metrics(
-                y_pred_postprocessed[..., indices],
-                y_postprocessed[..., indices],
-            )
+
+        for metric in self.metrics:
+            metric_name = getattr(metric, "name", metric.__class__.__name__)
+
+            for mkey, indices in self.metric_ranges_validation.items():
+                # for individual variables do not feature scale
+                if len(indices) == 1:
```

Comment:
If this is only about feature scaling, which it seems to be, this could simply be a boolean `bladibla` and one call `metric(...., feature_scale=bladibla)` instead.

Then it's also clear that `bladibla = mkey == "all"` and nothing more.

Reply:
hi I think @Rilwan-Adewoyin  would be able to explain this better as it's something that's used in patched energy

Reply:
tl;dr - for simplicity, the `if len(indices)==1` and `len(indices)>1` logic can be removed, the `elif mkey == "all"` logic can be removed. I would recommend set feature_scale=False and not pass in  feature_indices arg. 

Previously: 
- self.metric_ranges_validation holds mkeys, indices relating to weather variables grouped over pressure levels, and surface variables as a final group.
- in the prev code, when we calculate the loss over the groups (old line 272:275) we use post_processed values 
- I can't remember what the previous behaviour was regarding use of post_processed or non post processed
- When observing validation plots in mlflow, having feature_scaling True or False lead to changes in relative performance of model when looking at performance of these groups (indices) e.g. ModelA>ModelB w/ feature scaling and vice versa otherwise. So this was more useful when I was investigating the perf of different objective funcs. But I don't think it'd be a common usecase at this point so happy for it to be simplified as Jesper proposed. Plus this branch only has the three loss functions, none of which are multivariate so less usecase for configurability here.

Reply:
Thanks for the clarification, the changes have been encorporated